docs/docs/development/hardware-integration/soft-off-setup.mdx

```diff
@@ -0,0 +1,158 @@
+    <GpioKeyMatrix />
+  </TabItem>
+  <TabItem value="wakeup" label="Wakeup-only Key Switch">
+    <GpioKeyWakeup />
+  </TabItem>
+</Tabs>
+
+GPIO keys are defined using child nodes under the `gpio-keys` compatible node. Each child needs just one property defined:
+
+- The `gpios` property should be a [phandle-array](https://docs.zephyrproject.org/3.5.0/build/dts/phandles.html#zero-or-more-nodes-with-metadata-phandle-array-type) with a fully defined GPIO pin and with the correct pull up/down and active high/low flags set.
+
+<Tabs groupId="advanced-methods" defaultValue="direct" className="secrettabs">
+  <TabItem value="direct" label="Direct Pin">
+    <SidebandDirect />
+  </TabItem>
+  <TabItem value="matrix" label="Matrix Integrated Pin">
+    <SidebandMatrix />
+  </TabItem>
+  <TabItem value="wakeup" label="Wakeup-only Key Switch">
+    <Tabs groupId="kscan-type" defaultValue="mkscan" className="secrettabs">
+      <TabItem value="dkscan" label="Direct GPIO kscan">
+        <SidebandWakeupDirect />
+      </TabItem>
+      <TabItem value="mkscan" label="Matrix kscan"></TabItem>
+    </Tabs>
+  </TabItem>
+</Tabs>
+
+<Tabs groupId="advanced-methods" defaultValue="direct" className="secrettabs">
+  <TabItem value="direct" label="Direct Pin"></TabItem>
+  <TabItem value="matrix" label="Matrix Integrated Pin">
+    You also need to update the `zmk,kscan` chosen value to point to the new kscan instance:
+
+```dts
+/ {
+    chosen {
+        ...
+        zmk,kscan = &side_band_behavior_triggers;
+        ...
+    };
+};
+```
+
+    <SoftOffWaker />
+
+  </TabItem>
+  <TabItem value="wakeup" label="Wakeup-only Key Switch">
+    <Tabs groupId="kscan-type" defaultValue="mkscan" className="secrettabs">
+        <TabItem value="dkscan" label="Direct GPIO kscan"></TabItem>
+        <TabItem value="mkscan" label="Matrix kscan"><SoftOffWaker /></TabItem>
+    </Tabs>
+  </TabItem>
+</Tabs>
+
+Finally, we will list the `wakeup_scan` device in an additional configuration section so that the ZMK soft off process knows it needs to enable this device as part of the soft off processing so it can wake the keyboard from soft off when pressed:
+
+```dts
+/ {
+    soft_off_wakers {
+        compatible = "zmk,soft-off-wakeup-sources";
+        wakeup-sources = <&wakeup_scan>;
+    };
+};
+```
+
+Here are the properties for the node:
+
+- The `compatible` property for the node must be `zmk,soft-off-wakeup-sources`.
+- The `wakeup-sources` property is a [phandle array](../../config/index.md#devicetree-property-types) pointing to all the devices that should be enabled during the shutdown process to be sure they can later wake the keyboard.
+
+:::tip
+If you add your kscan to the `wakeup-sources` array, then your keyboard will wake upon pressing any key in your kscan. Essentially, this causes `&soft_off` to behave as one would expect `&sleep` to behave. If you choose to do so, then you can omit everything aside from the `soft_off_wakers` node.
```

Comment:
```suggestion
If you add your kscan to the `wakeup-sources` array, then your keyboard will wake upon pressing any key in your kscan. Essentially, this causes `&soft_off` to behave like a behavior that puts the keyboard in deep sleep. If you choose to do so, then you can omit everything aside from the `soft_off_wakers` node.
```

I haven't found a reference to `&sleep` in the page so I think it might not be intuitively clear what it would supposedly be for.